### PR TITLE
new(outputs): expose rule tags and event source in gRPC and json outputs

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area engine

**What this PR does / why we need it**:
1. Adds rule tags as a new field to the gRPC `outputs` service.
2. Adds rule tags and event source fields to the json outputs.
3. Updates the test suite to check that rule tags and event source are correctly retrievable through the gRPC api and through json outputs.

**Which issue(s) this PR fixes**:

Fixes https://github.com/falcosecurity/falco/issues/1647, https://github.com/falcosecurity/falco/issues/1713

**Does this PR introduce a user-facing change?**:

```release-note
new(outputs): expose rule tags and event source in gRPC and json outputs
```

### CODE REVIEW

**Review by leogr — COMMENTED** [review]
Just found a spacing issue (see the comment below), otherwise LGTM! 
Btw, great improvements, thank you! :+1: 

Note for reviewers: since the Lua code has been changed, to avoid a headache, make sure your local Falco's built is pointing to the correct Lua files :smiley_cat:

**Comment by leogr** [comment]
cc @Issif

**Review by leogr — APPROVED** [review]
/approve

**Review by mstemm — APPROVED** [review]

**Comment by leodido** [comment]
Late to the party...

Have you folks considered making the tags configurable?

The reason they were not there was also that, as far as I remember, the `tags` are not a mandatory part (while `output` is)  of the Falco rule definitions. 

Furthermore, a test covering such a scenario (no tags defined in the rule) would have been helpful to understand what's the resulting json in that case (empty array? other? valid json?).

File: userspace/falco/outputs.proto [hunk 1]
```diff
@@ -50,6 +50,5 @@ message response {
   string output = 5;
   map<string, string> output_fields = 6;
   string hostname = 7;
-  // TODO(leodido,fntlnz): tags not supported yet, keeping it for reference.
-  // repeated string tags = 8;
+  repeated string tags = 8;
```
Comment (leodido) [inline userspace/falco/outputs.proto:53]: Maybe this should have been `optional` in the protobuf def?

File: userspace/falco/outputs_grpc.cpp [hunk 2]
```diff
@@ -64,5 +64,9 @@ void falco::outputs::output_grpc::output(const message *msg)
 	auto host = grpc_res.mutable_hostname();
 	*host = m_hostname;
 
+	// tags
+	auto tags = grpc_res.mutable_tags();
+	*tags = {msg->tags.begin(), msg->tags.end()};
```
Comment (leodido) [inline userspace/falco/outputs_grpc.cpp:69]: Let's check what we end up with when a Falco rule does **not** have tags defined.

Do we obtain `...,"tags":[],...` or not? I'd prefer a clean JSON, but that's my personal preference only :)

**Review by leodido — COMMENTED** [review]
Great job Jason!

Left two comments

**Comment by Issif** [comment]
I'm really of this PR, when falco 0.30.0 is out, I'll update the format of json in Falcosidekick either 👍 .

File: userspace/falco/outputs.proto (hunk 1 above)
Comment (leogr) [reply userspace/falco/outputs.proto:53]: AFAIK, a `repeated` field is inherently `optional`.

Moreover, the protobuf compiler does not accept both modifiers simultaneously, for example:

```
optional repeated string tags = 8;
```

will produce a syntax error:

```
optional repeated string tags = 8;
```

Unless I'm missing something, we cannot add `optional` (and we don't need it).

File: userspace/falco/outputs_grpc.cpp (hunk 2 above)
Comment (leogr) [reply userspace/falco/outputs_grpc.cpp:69]: I don't have any preferences. Anyway, I agree with @leodido that a test covering such a scenario (especially for the JSON output format) would be really helpful to clarify the expected result.

File: userspace/falco/outputs.proto (hunk 1 above)
Comment (jasondellaluce) [reply userspace/falco/outputs.proto:53]: I endorse the comment of @leogr. From my understanding, the behavior of `repeated` is quite similar to `optional` already.

File: userspace/falco/outputs_grpc.cpp (hunk 2 above)
Comment (jasondellaluce) [reply userspace/falco/outputs_grpc.cpp:69]: Thank you for your review! I opened a PR to address these concerns, so we can eventually move this conversation there: https://github.com/falcosecurity/falco/pull/1733

**Pull request merged**